pxtblocks/plugins/flyout/blockInflater.ts

```diff
@@ -74,6 +68,46 @@ export class MultiFlyoutRecyclableBlockInflater extends Blockly.BlockFlyoutInfla
         block.moveBy(-xy.x, -xy.y);
         block.addClass(HIDDEN_CLASS_NAME);
         block.setDisabledReason(true, HIDDEN_CLASS_NAME);
+
+        // blockly deletes all of the potential variables in the flyout after recycling blocks.
+        // this can end up deleting our cached blocks if one of the referenced variables was
+        // promoted from a potential variable to a real one, since it will detect all of the
+        // cached blocks as referencing that variable. we can remap any variable that exists
+        // as both a potential and non-potential variable to a new one with the same name to
+        // prevent its usages from being deleted
+        const potentialVariableMap = block.workspace.getPotentialVariableMap();
+
+        if (potentialVariableMap) {
+            const allBlocks = [block].concat(block.getChildren(false));
+
+            // loop over all child variable fields
+            for (const block of allBlocks) {
```

Comment:
nit: could we rename this to childBlock or something, so it doesn't conflict with the block passed into the `recycleBlock` function?